framework/src/meshgenerators/SideSetsGeneratorBase.C

```diff
@@ -64,6 +64,16 @@ SideSetsGeneratorBase::setup(MeshBase & mesh)
   _fe_face = FEBase::build(dim, fe_type);
   _qface = libmesh_make_unique<QGauss>(dim - 1, FIRST);
   _fe_face->attach_quadrature_rule(_qface.get());
+
+  // Change this when we have more fine-grained control over advertising what we need need and how
+  // we satisfy those needs. For now we know we need to have neighbors...and we do have an explicit
+  // `find_neighbors` call...but we don't have a `neighbors_found` API and it seems off to do:
+  //
+  // if (!mesh.is_prepared())
+  //   mesh.find_neighbors()
+
+  if (!mesh.is_prepared())
+    mesh.prepare_for_use();
```

Comment:
I was wondering why the mesh generator needs to call `mesh.prepare_for_use`?  Ideally, `MeshSetupCompleteAction` is the right guy to manage this.

Reply:
This mesh generator requires that elements have properly had their neighbors set. See https://github.com/idaholab/moose/issues/15823

Reply:
I added to the code comment, about how we need neighbors properly set, that the interested party should look at #15823